src/network_models/power_flow_evaluation.jl

```diff
@@ -594,39 +589,75 @@ _get_pf_result(::Type{PowerFlowVoltageAngle}, pf_data::PFS.PowerFlowData) =
+    PFS.get_arc_activepower_flow_to_from(pf_data)
 _get_pf_result(::Type{PowerFlowLossFactors}, pf_data::PFS.PowerFlowData) =
     PFS.get_loss_factors(pf_data)
 _get_pf_result(::Type{PowerFlowVoltageStabilityFactors}, pf_data::PFS.PowerFlowData) =
     PFS.get_voltage_stability_factors(pf_data)
 
-_get_pf_lookup(::Type{<:PSY.Bus}, pf_data::PFS.PowerFlowData) = PFS.get_bus_lookup(pf_data)
-_get_pf_lookup(::Type{<:PSY.Branch}, pf_data::PFS.PowerFlowData) =
-    PFS.get_branch_lookup(pf_data)
+function calculate_aux_variable_value!(container::OptimizationContainer,
+    key::AuxVarKey{T, <:PSY.ACBus}, # TODO: does this work ok with DCBuses, too?
+    ::PSY.System,
+    pf_e_data::PowerFlowEvaluationData{<:PFS.PowerFlowData},
+) where {T <: PowerFlowAuxVariableType}
+    @debug "Updating $key from PowerFlowData"
+    pf_data = get_power_flow_data(pf_e_data)
+    src = _get_pf_result(T, pf_data)
+    lookup = PFS.get_bus_lookup(pf_data)
+    dest = get_aux_variable(container, key)
+    for component_id in axes(dest, 1)  # these are bus numbers
+        dest[component_id, :] = src[lookup[component_id], :]
+    end
+    return
+end
 
 function calculate_aux_variable_value!(container::OptimizationContainer,
     key::AuxVarKey{T, U},
-    system::PSY.System, pf_e_data::PowerFlowEvaluationData{<:PFS.PowerFlowData},
-) where {T <: PowerFlowAuxVariableType, U}
+    ::PSY.System,
+    pf_e_data::PowerFlowEvaluationData{<:PFS.PowerFlowData},
+) where {T <: PowerFlowAuxVariableType, U <: PSY.Branch}
     @debug "Updating $key from PowerFlowData"
     pf_data = get_power_flow_data(pf_e_data)
     src = _get_pf_result(T, pf_data)
-    lookup = _get_pf_lookup(U, pf_data)
     dest = get_aux_variable(container, key)
-    for component_id in axes(dest, 1)  # these are bus numbers or branch names
-        dest[component_id, :] = src[lookup[component_id], :]
+    nrd = PFS.get_network_reduction_data(pf_data)
+    arc_lookup = PFS.get_arc_lookup(pf_data)
+    # PERF: could pre-compute a Dict of branch type to arcs, then intersect the arcs
+    # for the type U with the keys of the branch maps.
+    for (arc, br) in nrd.direct_branch_map
+        if br isa U # always a concrete class, so same as: typeof(br) == U
+            name = PSY.get_name(br)
+            arc_ix = arc_lookup[arc]
+            dest[name, :] = src[arc_ix, :]
+        end
+    end
+    for (arc, parallel_brs) in nrd.parallel_branch_map # parallel_brs is Set{ACTransmission}
+        sample_line = first(parallel_brs)
+        impedance = PSY.get_r(sample_line) + im * PSY.get_x(sample_line)
+        n_branches = length(parallel_brs)
+        for br in parallel_brs
+            if br isa U
+                @assert T <: PowerFlowLineFlowType "Only PowerFlowLineFlowType aux vars " *
+                                                   "can be used for parallel branches: got $T"
+                @assert PSY.get_r(br) + im * PSY.get_x(br) == impedance "All parallel " *
```

Comment:
is this properly an `@assert` or an error? I think it's only an `@assert` if we aim to make it impossible to end up in this situation using public interface